internal/transport/http_util_test.go

```diff
@@ -215,6 +218,37 @@ func (s) TestParseDialTarget(t *testing.T) {
 	}
 }
 
+type badNetworkConn struct {
+	net.Conn
+}
+
+func (c *badNetworkConn) Write([]byte) (int, error) {
+	return 0, io.EOF
+}
+
+func TestWriteBadConnection(t *testing.T) {
+	data := []byte("test_data")
+	writeBufferSize := len(data) - 1
+	writer := newBufWriter(&badNetworkConn{}, writeBufferSize, getWriteBufferPool(writeBufferSize))
+
+	syncCh := make(chan struct{})
+	var err error
+	go func() {
+		_, err = writer.Write(data)
+		close(syncCh)
+	}()
+
+	select {
+	case <-time.After(time.Second):
+		t.Fatalf("Write() did not return in time")
+	case <-syncCh:
+	}
+
+	if !errors.Is(err, io.EOF) {
+		t.Fatalf("Write() did not return an error or returned unexpeced one: err=%v", err)
```

Comment:
nit: Typo in the spelling of `unexpected`. However, I would recommend changing the error message to match the Google style guide that gRPC follows. See https://google.github.io/styleguide/go/decisions#level-of-detail and

```go
t. Fatalf("Write() = %v, want error presence = %v", err, io.EOF)
```